packages/main/src/MenuItem.ts

```diff
@@ -201,6 +227,25 @@ class MenuItem extends ListItem implements IMenuItem {
 	@i18n("@ui5/webcomponents")
 	static i18nBundle: I18nBundle;
 
+	/**
+	 * Defines whether `ui5-menu-item` is in selected state.
+	 *
+	 * **Note:** selected state is only taken into account when `ui5-menu-item` is added to `ui5-menu-item-group` with `itemSelectionMode` other than `None`.
+	 * **Note:** A selected `ui5-menu-item` have selection mark displayed ad its end.
+	 * @default false
+	 * @public
+	 * @since 2.4.0
```

Comment:
since 2.5.0

why do we need setter and getter? they do synchronous property change and it should be used in very edge cases.  

overall why do we need isSelected property which leads to inconsistency with other components that provide selection? what is the difference between selected and isSelected overall?

Reply:
Setter is needed because if you select item that is inside a MenuItemGroup with `Single` selection mode, other selected items should be deselected automatically by the group. Here we fire `item-selection` event that is handled by group and if it is necessary (selection mode is `Single`) the group unselects other items.

As I already wrote as answer of Ilhan's comment, the **MenuItem** extends **ListItem** that also have `selection` property, but it is related to totally different behavior, so selection here is different and we need different property for this. `selected` in ListItem does this: 

![image](https://github.com/user-attachments/assets/a987dd9d-06fa-4bb1-8beb-9d30265b5762)

and `isSelected` in MenuItem t=does this: 

![image](https://github.com/user-attachments/assets/0d728fbe-74a0-424c-a9a8-d851af427205)

and at the same time we can have "original" ListItem `selected` state in Menu like here:

![image](https://github.com/user-attachments/assets/9b7b29b8-fdd3-457b-b531-e4bf0fb2cf9b)